File: app/src/main/java/com/webianks/hatkemessenger/adapters/SingleGroupAdapter.java
package com.webianks.hatkemessenger.adapters;

import android.content.Context;
import android.database.Cursor;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.amulyakhare.textdrawable.TextDrawable;
import com.webianks.hatkemessenger.R;
import com.webianks.hatkemessenger.utils.ColorGeneratorModified;
import com.webianks.hatkemessenger.utils.Helpers;

/**
 * Created by R Ankit on 25-12-2016.
 */

public class SingleGroupAdapter extends RecyclerView.Adapter<SingleGroupAdapter.MyViewHolder> {

    private ColorGeneratorModified generator;
    private Context context;
    private Cursor dataCursor;
    private int color;

    public SingleGroupAdapter(Context context, Cursor dataCursor, int color) {

        this.context = context;
        this.dataCursor = dataCursor;
        this.color = color;

        if (color == 0)
            generator = ColorGeneratorModified.MATERIAL;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View view = inflater.inflate(R.layout.single_sms_detailed, parent, false);
        return new MyViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {

        dataCursor.moveToPosition(position);
        holder.message.setText(dataCursor.getString(dataCursor.getColumnIndexOrThrow("body")));

        long time = dataCursor.getLong(dataCursor.getColumnIndexOrThrow("date"));
        holder.time.setText(Helpers.getDate(time));

        String name = dataCursor.getString(dataCursor.getColumnIndexOrThrow("address"));
        String firstChar = String.valueOf(name.charAt(0));

        if (color == 0){
            if (generator!=null)
                color = generator.getColor(name);
        }

        TextDrawable drawable = TextDrawable.builder().buildRound(firstChar, color);
        holder.image.setImageDrawable(drawable);


    }

    public void swapCursor(Cursor cursor) {
        if (dataCursor == cursor) {
            return;
        }
        this.dataCursor = cursor;
        if (cursor != null) {
            this.notifyDataSetChanged();
        }
    }

    @Override
    public int getItemCount() {
        return (dataCursor == null) ? 0 : dataCursor.getCount();
    }


    class MyViewHolder extends RecyclerView.ViewHolder {

        private TextView message;
        private ImageView image;
        private TextView time;

        MyViewHolder(View itemView) {
            super(itemView);

            message =  itemView.findViewById(R.id.message);
            image =  itemView.findViewById(R.id.smsImage);
            time =  itemView.findViewById(R.id.time);

        }

    }
}

File: app/src/main/java/com/webianks/hatkemessenger/adapters/SingleGroupAdapter.kt
package com.webianks.hatkemessenger.adapters

import android.content.Context
import android.database.Cursor
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.amulyakhare.textdrawable.TextDrawable
import com.webianks.hatkemessenger.R
import com.webianks.hatkemessenger.adapters.SingleGroupAdapter.MyViewHolder
import com.webianks.hatkemessenger.utils.ColorGeneratorModified
import com.webianks.hatkemessenger.utils.Helpers

/**
 * Created by R Ankit on 25-12-2016.
 */
class SingleGroupAdapter(private val context: Context,
                         private var dataCursor: Cursor?,
                         private var color: Int,
                         private  val savedContactName: String?) : RecyclerView.Adapter<MyViewHolder>() {

    private lateinit var generator: ColorGeneratorModified

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        val view = inflater.inflate(R.layout.single_sms_detailed, parent, false)
        return MyViewHolder(view)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        dataCursor!!.moveToPosition(position)
        holder.message.text = dataCursor!!.getString(dataCursor!!.getColumnIndexOrThrow("body"))
        val time = dataCursor!!.getLong(dataCursor!!.getColumnIndexOrThrow("date"))
        holder.time.text = Helpers.getDate(time)
        val name = dataCursor!!.getString(dataCursor!!.getColumnIndexOrThrow("address"))
        val firstChar = savedContactName?.get(0).toString() ?: name[0].toString()

        if (color == 0)
            color = generator.getColor(name)

        val drawable = TextDrawable.builder().buildRound(firstChar, color)
        holder.image.setImageDrawable(drawable)
    }

    fun swapCursor(cursor: Cursor?) {
        if (dataCursor === cursor) {
            return
        }
        dataCursor = cursor
        if (cursor != null) {
            notifyDataSetChanged()
        }
    }

    override fun getItemCount(): Int {
        return if (dataCursor == null) 0 else dataCursor!!.count
    }

    inner class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val message: TextView = itemView.findViewById(R.id.message)
        val image: ImageView = itemView.findViewById(R.id.smsImage)
        val time: TextView = itemView.findViewById(R.id.time)
    }

    init {
        if (color == 0) generator = ColorGeneratorModified.MATERIAL!!
    }
}